examples/data_preprocessing/police_killing/preprocessing_with_kb.ipynb

```diff
@@ -0,0 +1,1883 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_rule2id(intersection_fe_and_samples: Set) -> Dict:\n",
+    "    rule2rule_id = {}\n",
+    "    rule_id = 0\n",
+    "    for name in intersection_fe_and_samples: \n",
+    "        rule2rule_id[name] = rule_id\n",
+    "        rule_id += 1\n",
+    "\n",
+    "    return rule2rule_id\n",
+    "\n",
+    "rule2rule_id = get_rule2id(intersection_fe_and_samples)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "There are 2654 rules.\n",
+      "\n",
+      "The first rules of the rule2rule_id dictionary look like this:\n",
+      "{'John Toles': 0, 'Martin Whittaker': 1, 'Ryan Stokes': 2, 'Javier Garcia': 3, 'Benjamin Ortiz': 4, 'David Lepine': 5}\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"There are {len(rule2rule_id)} rules.\")\n",
+    "print(\"\\nThe first rules of the rule2rule_id dictionary look like this:\")\n",
+    "print(dict(islice(rule2rule_id.items(), 6)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Secondly, we create a dictionary assigning all rules to their label. There are only two classes (someone was killed by the police or was not killed by the police). Since there are no rules indicating that someone was **not** killed by the police, all rules indicate the positive class 1. Therefore, all values of the rule2label dictionary, containing the rule IDs as keys, can be set to 1."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Actually, we don't even need these dicts anymore for this preprocesing. Do I have to keep them or can I just delete that?**\n",
+    "(it's just there because there are also rule2label and label2label_id in the other tutorials)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rule2label = {rule_id: 1 for rule_id in rule2rule_id.values()}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Thirdly, we create a label2label_id dictionary. As there are only two classes, this can be done manually."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "label2label_id = {\"negative\":0, \"positive\":1}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3 Build the T matrix (rules x classes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The rows of the T matrix are the rules and the columns the classes. The T matrix is one-hot encoded. (1 for a rule and its corresponding class.) It will be imported from the data_preprocessing folder of Knodle examples, since the same function can be used in several preprocessing tutorials. "
```

Comment:
```suggestion
    "The rows of the T matrix are the rules and the columns are the classes. The T matrix is one-hot encoded (1 for a rule and its corresponding class). It will be imported from the data_preprocessing folder of Knodle examples, since the same function can be used in several preprocessing tutorials. "
```